nc/source/ecs/ECS.cpp

```diff
@@ -0,0 +1,134 @@
+#include "ECS.h"
+#include "ecs/ECSImpl.h"
+#include "component/Renderer.h"
+#include "component/PointLight.h"
+
+namespace nc
+{
+ecs::ECSImpl* ECS::m_impl = nullptr;
+
+void ECS::RegisterImpl(ecs::ECSImpl* impl)
```

Comment:
Case all instances of "ECS" like "Ecs", Microsoft guidelines we follow at work that I like are:
Acronyms with two letters have both letters capitalized.
Acronyms with more than two letters just capitalize the first letter.
It makes it easier to read in cases like "ECSImpl" -> is it "EC Simpl" "E CS Impl" etc.